crates/linear_algebra/src/vector.rs

```diff
@@ -0,0 +1,161 @@
+        Point::from(self.inner)
+    }
+}
+
+impl<Frame, const DIMENSION: usize, Scalar> Framed<Frame, SVector<Scalar, DIMENSION>>
+where
+    Scalar: nalgebra::Scalar + ComplexField<RealField = Scalar>,
+{
+    pub fn normalize(&self) -> Self {
+        Self::wrap(self.inner.normalize())
+    }
+
+    pub fn try_normalize(&self, min_norm: Scalar) -> Option<Self> {
+        Some(Self::wrap(self.inner.try_normalize(min_norm)?))
+    }
+
+    pub fn cap_magnitude(&self, max: Scalar) -> Self {
+        Self::wrap(self.inner.cap_magnitude(max))
+    }
+
+    pub fn unscale(&self, real: Scalar) -> Self {
+        Self::wrap(self.inner.unscale(real))
+    }
+
+    pub fn norm(&self) -> Scalar {
+        self.inner.norm()
+    }
+
+    pub fn norm_squared(&self) -> Scalar {
+        self.inner.norm_squared()
+    }
+
+    pub fn dot(&self, rhs: Self) -> Scalar {
+        self.inner.dot(&rhs.inner)
+    }
+
+    pub fn angle(&self, rhs: Self) -> Scalar {
+        self.inner.angle(&rhs.inner)
+    }
+
+    pub fn component_mul(&self, rhs: Self) -> Self {
+        Self::wrap(self.inner.component_mul(&rhs.inner))
+    }
+}
+
+impl<Frame, const DIMENSION: usize, Scalar> Framed<Frame, SVector<Scalar, DIMENSION>>
+where
+    Scalar: nalgebra::Scalar + ComplexField<RealField = Scalar> + Signed,
+{
+    pub fn abs(&self) -> Self {
+        Self::wrap(self.inner.abs())
+    }
+}
+
+impl<Frame, Scalar> Framed<Frame, SVector<Scalar, 2>>
+where
+    Scalar: nalgebra::Scalar + Copy,
+{
+    pub fn x(&self) -> Scalar {
+        self.inner.x
+    }
+
+    pub fn y(&self) -> Scalar {
+        self.inner.y
+    }
+}
+
+impl<Frame, Scalar> Framed<Frame, SVector<Scalar, 2>>
+where
+    Scalar: nalgebra::Scalar + Zero + One + Copy,
+{
+    pub fn x_axis() -> Self {
+        Self::wrap(*SVector::x_axis())
+    }
+
+    pub fn y_axis() -> Self {
+        Self::wrap(*SVector::y_axis())
+    }
+}
+
+impl<Frame, Scalar> Framed<Frame, SVector<Scalar, 3>>
+where
+    Scalar: nalgebra::Scalar + Copy,
+{
+    pub fn x(&self) -> Scalar {
+        self.inner.x
+    }
+
+    pub fn y(&self) -> Scalar {
+        self.inner.y
+    }
+
+    pub fn z(&self) -> Scalar {
+        self.inner.z
+    }
+}
+
+impl<Frame, Scalar> Framed<Frame, SVector<Scalar, 3>>
+where
+    Scalar: nalgebra::Scalar + Zero + One + Copy,
+{
+    pub fn x_axis() -> Self {
+        Self::wrap(*SVector::x_axis())
+    }
+
+    pub fn y_axis() -> Self {
+        Self::wrap(*SVector::y_axis())
+    }
+
+    pub fn z_axis() -> Self {
+        Self::wrap(*SVector::z_axis())
+    }
+}
+
+impl<Frame, Scalar> Framed<Frame, SVector<Scalar, 3>>
+where
+    Scalar: nalgebra::Scalar,
+{
+    pub fn xy(&self) -> Vector2<Frame, Scalar> {
+        Vector2::wrap(self.inner.xy())
+    }
+
+    pub fn xz(&self) -> Vector2<Frame, Scalar> {
+        Vector2::wrap(self.inner.xz())
+    }
+
+    pub fn yz(&self) -> Vector2<Frame, Scalar> {
+        Vector2::wrap(self.inner.yz())
+    }
+}
```

Comment:
You split this from the `x`, `y`, `z` block above, but they are in the same block for the `Point` implementation, why?

Reply:
I was just lazy... I first tried to have minimal trait bounds. But because of readability I just assumed some basic bounds. To further increase readability, I now changed it to merge even more basic constraints. like `Copy + Zero + One` etc... -> YAGNI